File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    bool isVowel(char c) {
        return c=='a' || c=='e' || c=='i' || c=='o' || c=='u';
    }

    int maxVowels(string s, int k) {
        int i = 0, count = 0, mx = 0;
        for (int j = 0; j < s.size(); j++) {
            if (isVowel(s[j])) {
                count++;
            }
            if (j - i + 1 > k) {
                if (isVowel(s[i])) {
                    count--;
                }
                i++;
            }
            if (j - i + 1 == k) {
                mx = max(mx, count);
            }
        }

        return mx;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {

    public static boolean isvow(char a) {
        if (a == 'a' || a == 'e' || a == 'i' || a == 'o' || a == 'u' ||
            a == 'A' || a == 'E' || a == 'I' || a == 'O' || a == 'U') {
            return true;
        }
        return false;
    }

    public int maxVowels(String s, int k) {
        int i = 0;
        int j = 0;
        int count = 0;
        int max = 0;

        while (j < s.length()) {

            if (isvow(s.charAt(j))) {
                count++;
            }

            if (j - i + 1 > k) {
                if (isvow(s.charAt(i))) {
                    count--;
                }
                i++;
            }

            if (j - i + 1 == k) {
                max = Math.max(max, count);
            }

            j++;
        }
        return max;
    }
}
